app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt

```diff
@@ -178,6 +170,14 @@ class RecentlyPlayedFragmentTest {
 
   @Test
   fun testRecentlyPlayedTestActivity_sectionDividerIsNotDisplayed() {
+    storyProgressTestHelper.markRecentlyPlayedForFractionsStory0Exploration0(
```

Comment:
Reminder to please make sure you add tests that have parity with the domain layer tests regarding all the different edge cases of how the list can be merged/computed.

Reply:
> @veena14cs Can you add screenshots to upcoming topics to the description?
> 
> ### Problem 1
> 1. In `HomeFragmentPresenter` using `storyProgressTestHelper.markFullProgressForAllTopics(ProfileId.newBuilder().setInternalId(internalProfileId).build(), false)` we can mark all topics as finished.
> 2. Add the above line and create a new profile.
> 3. Open this new profile, you will see topics. But instead it should show upcoming topics right?
> 4. If you will open each topic it will show that the progress has been finished for all topics.
> 
> ### Problem 2
> 1. Create a new profile.
> 2. Open `First Topic` -> `First Story`
> 3. Open `First Topic` -> `Second Story`
> 4. Now both the above stories should come as part of recently played and we will suggest `Second Topic` as par of `suggestions`.
> 5. As per our discussion in this case the order in Home Screen should be
> 
> * `First Story` as part of Recently Played,
> * `Fifth Exploration` as part of Suggestions
> * and then `Second Story` as part of Recently Played
> 
> But instead its showing
> 
> * `First Story` as part of Recently Played,
> * `Second Story` as part of Recently Played
> * and then `Fifth Exploration` as part of Suggestions

- Marking full progress for all topics done. It shows Coming soon topic.
- 2nd Problem you mentioned does not match with the current Implementation as per @Ben discussion on https://github.com/oppia/oppia-android/pull/2272#discussion_r563187413.
- The computation of recommended stories is changed to what we had in Design Doc. I have created a method that retrieves dependent topics that user needs to learn as prerequisite.  
- I have added testcase that best defines the behavior we discussed regarding showing recommended stories over completed topic. You can look at the testcase `testHomeActivity_markStory0DonePlayStory1FirstTestTopic_playRatios_storiesForYouIsCorrect`